tests/functional/contribute/test_signup.py

```diff
@@ -0,0 +1,62 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import pytest
+from selenium.common.exceptions import TimeoutException
+
+from ...pages.contribute.signup import ContributeSignUpPage
+
+
+@pytest.mark.sanity
+@pytest.mark.nondestructive
+def test_toggle_category_select(base_url, selenium):
+    page = ContributeSignUpPage(base_url, selenium).open()
+    category = page.contributor_categories[0]  # Select coding category
+    area = page.category_areas[0]  # Coding category areas
+    category.select_category
+    assert area.changes_to_required_field
+    assert area.is_displayed
+    category = page.contributor_categories[4]  # Select helping category
+    category.select_category
+    area.wait_to_collapse()
+    assert not area.is_required
+    assert not area.is_displayed
+
+
+def test_successful_sign_up_coding(base_url, selenium):
+    page = ContributeSignUpPage(base_url, selenium).open()
+    category = page.contributor_categories[0]  # Select coding category
+    area = page.category_areas[0]  # Coding category areas
+    category.select_category
+    assert area.changes_to_required_field
+    assert area.is_displayed
+    area.select_area('Firefox')
+    page.type_name('Automated test')
+    page.type_email('noreply@mozilla.com')
+    page.select_country('United Kingdom')
+    page.select_text_format()
+    page.accept_privacy_policy()
+    page.click_start_contributing()
+    assert '/{0}'.format('?c=coding') in selenium.current_url, 'Category is not in URL'
+
+
+@pytest.mark.skipif(reason='https://bugzilla.mozilla.org/show_bug.cgi?id=1221526')
+def test_successful_sign_up_helping(base_url, selenium):
```

Comment:
As above, these tests could be combined using parameterisation, but I'd at least make the category selection be by label instead of index.